### PR TITLE
Update plugin for Pico 1.0.0

### PR DESCRIPTION
**Pull request by theshka**
* [New] You can disable the plugin in `config/config.php` by
using the following:  `$config['PicoXMLSitemap.enabled']=false;`
* [Changed] Extend from new `AbstractPicoPlugin` class
* [Changed] Use of new hook-names
* [Changed] Expanded inline documentation
* [Changed] Updated `README.md`, note new query string url `?sitemap.xml`
* [Changed] Rename filename & classname to `PicoXMLSitemap.php` and `PicoXMLSiteMap`, respectively
* [Removed] `__construct` method, directly set `is_sitemap` to `false` when declaring
* [Removed] Unused `dependsOn` array
* [Removed] Closing `</lastmod>` tag when no date is available

### CODE REVIEW

File: PicoXMLSitemap.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+*/
+class PicoXMLSitemap extends AbstractPicoPlugin
+{
+    /**
+    * This plugin is enabled by default?
+    *
+    * @see AbstractPicoPlugin::$enabled
+    * @var boolean
+    */
+    protected $enabled = false;
```
Comment (PhrozenByte) [inline PicoXMLSitemap.php]: The addon has no required config, so it's imho reasonable to enable the plugin by default.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 
-* Place the pico_sitemap.php into your plugins directory.
-* Browse to sitemap.xml in your site root.
-* Take a break, your work is done.
+* Place the `PicoXMLSitemap.php` into your `plugins` directory.
 
-[1]: http://pico.dev7studios.com/
+* Place `$config['PicoXMLSitemap.enabled'] = true;` in your `config/config.php`
+
+* Browse to `http://yoursite.com/?sitemap.xml`
```
Comment (PhrozenByte) [inline README.md]: A sitemap is only recognized when it is available through `http://example.com/sitemap.xml`, so this plugin always requires enabled URL rewriting (or at least a single rewrite rule for the `sitemap.xml`) to work.

Comment (theshka) [inline README.md]: Why would the sitemap only be accessible at the root, and with URL rewriting @PhrozenByte ? I submitted a test sitemap to Google, and they will accept a sitemap at the url `http://example.com/?sitemap.xml` I couldn't see any mention of the sitemap not working at this location here either: http://www.sitemaps.org/protocol.html#location :confused: Care to elaborate?

Comment (PhrozenByte) [inline README.md]: Hmm... I always thought that a sitemap must be accessible through `sitemap.xml` to be recognized by search engines (as e.g. `robots.txt`). The docs you've provided also talks just about `sitemap.xml`, not "any other URL you provide" or something similar. Maybe Google is just a little less restrictive on this.

File: PicoXMLSitemap.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    /**
+    * This plugin depends on ...
+    *
+    * @see AbstractPicoPlugin::$dependsOn
+    * @var string[]
+    */
+    protected $dependsOn = null;
```
Comment (PhrozenByte) [inline PicoXMLSitemap.php]: Can be removed :wink:

File: PicoXMLSitemap.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+*/
+class PicoXMLSitemap extends AbstractPicoPlugin
+{
+    /**
+    * This plugin is enabled by default?
+    *
+    * @see AbstractPicoPlugin::$enabled
+    * @var boolean
+    */
+    protected $enabled = true;
```
Comment (PhrozenByte) [inline PicoXMLSitemap.php]: This is the default of `AbstractPicoPlugin` and can thus be removed

File: PicoXMLSitemap.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            header('Content-Type: application/xml; charset=UTF-8');
+            //XML Start
+            $xml = '<?xml version="1.0" encoding="UTF-8"?><urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">';
+            //Page loop
+            foreach( $pages as $page ){
+                //Page URL
+                $xml .= '<url><loc>'.$page['url'].'</loc>';
+                //Page date/last modified
+                if(!empty($page['date'])){
+                    $xml .= '<lastmod>'.$page['date'].'</lastmod>';
```
Comment (PhrozenByte) [inline PicoXMLSitemap.php]: Why don't you use `date('c', $page['time'])` instead?

**Pull request merged**